content/blog/2024-07-31-valkey-8-0-0-rc1.md

```diff
@@ -0,0 +1,126 @@
+date= 2024-07-31 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Delivering significant improvements in throughput, scalability, and observability for modern data-intensive applications"
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "pingxie" ]
++++
+
+# Valkey 8.0: Unleashing Unprecedented Performance and Reliability
+
+The Valkey team is proud to unveil the first release candidate of Valkey 8.0,
+Valkey 8.0 RC1, a major update designed to enhance performance, reliability,
+and observability for both standalone setups and clusters of various sizes.
+
+## Performance
+
+Valkey 8.0 features asynchronous I/O threading, allowing the main thread and
+I/O threads to operate concurrently. This enhancement significantly increases
+throughput, achieving up to 1.2 million Queries Per Second (QPS) on AWS's r7g
+platform, compared to the previous limit of 380K QPS.
+
+Valkey 8.0 focuses on intelligent optimization, not just raw speed. By fine-
+tuning CPU core utilization, minimizing command latency, and incorporating
+Memory Access Amortization (MAA) to streamline dictionary operations, Valkey
+8.0 delivers an in-memory data store that is both efficient and responsive.
+
+- **Asynchronous I/O Threading**: Enables parallel processing of commands and
+  I/O operations, maximizing throughput and minimizing bottlenecks.
+- **Optimized CPU Core Utilization**: Distributes I/O tasks across multiple
+  cores, reducing idle time and maximizing computational power.
+- **Memory Access Amortization (MAA)**: Optimizes memory access patterns by
+  prefetching frequently accessed data to minimize CPU cache misses, reducing
+  memory accesses required for dictionary operations.
+
+## Reliability
+
+Cluster scaling operations, particularly re-sharding and slot migrations, have
+historically been delicate. Valkey 8.0 improves reliability and minimizes
+disruptions with the following enhancements:
+
+- **Automatic Failover for Empty Shards**: New shards that start empty, owning
+  no slots, now benefit from automatic failover. This ensures high availability
+  from the start of the scaling process.
+- **Replication of Slot Migration States**: All `CLUSTER SETSLOT` commands are
+  now replicated synchronously on replicas before execution on the primary. This
+  guarantees a consistent slot migration state across the shard, preventing data
+  loss or inconsistencies if the primary fails.
+- **Logical Link Recovery**: In the event of a failover, Valkey 8.0 automatically
+  repairs the logical links between source and target nodes. This ensures requests
+  are correctly routed to the new primary in the target shard, maintaining cluster
+  integrity and availability.
```

Comment:
This point still confuses, I think saying there is a "logical link" between source and target is confusing because it implies a TCP link or a clusterbus link. We have `cluster links` for example. Maybe **Slot Migration State recovery** instead?